src/hooks/useOnyx.ts

```diff
@@ -0,0 +1,83 @@
+import {useMemo} from 'react';
+import {useOnyx as originalUseOnyx} from 'react-native-onyx';
+import type {OnyxCollection, OnyxEntry, OnyxKey, OnyxValue, UseOnyxOptions} from 'react-native-onyx';
+import {useSearchContext} from '@components/Search/SearchContext';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type {SearchResults} from '@src/types/onyx';
+
+type OriginalUseOnyx = typeof originalUseOnyx;
+type OriginalUseOnyxReturnType = ReturnType<OriginalUseOnyx>;
+
+const getDataByPath = (data: SearchResults['data'], path: string) => {
+    // Handle prefixed collections
+    for (const collection of Object.values(ONYXKEYS.COLLECTION)) {
+        if (path.startsWith(collection)) {
+            const key = `${collection}${path.slice(collection.length)}`;
+            return data?.[key as keyof typeof data];
+        }
+    }
+
+    // Handle direct keys
+    return data?.[path as keyof typeof data];
+};
+
+// Helper function to get key data from snapshot
+const getKeyData = <TKey extends OnyxKey, TReturnValue>(snapshotData: SearchResults, key: TKey, initialValue?: TReturnValue): TReturnValue => {
+    if (key.endsWith('_')) {
+        // Create object to store matching entries
+        const result: OnyxCollection<TKey> = {};
+        const prefix = key;
+
+        // Get all keys that start with the prefix
+        Object.entries(snapshotData?.data ?? {}).forEach(([dataKey, value]) => {
+            if (!dataKey.startsWith(prefix)) {
+                return;
+            }
+            result[dataKey] = value as OnyxEntry<TKey>;
+        });
+        return (Object.keys(result).length > 0 ? result : initialValue) as TReturnValue;
+    }
+    return (getDataByPath(snapshotData?.data, key) ?? initialValue) as TReturnValue;
+};
+
+/**
+ * Custom hook for accessing and subscribing to Onyx data with search snapshot support
+ */
+const useOnyx: OriginalUseOnyx = (key, options, dependencies) => {
+    const {isOnSearch, currentSearchHash} = useSearchContext();
```

Comment:
hey, just noticed this - I did a lot of search coding previously.

My main worry is that this new `useOnyx` hook will re-render quite often since it consumes search context. How will we know whether some component in future should use normal `useOnyx` or the new for attaching? Other views that are not necessarily connected to Search might now also re-render when search stuff changes - have we discussed how this will impact the amount of rerenders and performance?